### DISCUSSION TITLE
Need some help for Layouts - single-shot process closes layout panel instead of keeping it open..

### DISCUSSION THREAD

**Discussion by pm5k** [kovidgoyal/kitty#3646] [Q&A] [answered]
I am trying to configure my startup session to launch several pre-configured spaces, but it seems like processes that run and stop will end up being closed by Kitty, instead of remaining open. I was wondering how I could ensure that this does not happen. I am currently testing this behaviour with the following startup session file:

```sh
launch cat ~/hello.txt
launch zsh
layout tall
```

This runs `cat` in the left panel/window and immediately closes that panel, while what is supposed to be the right panel launches `zsh` and remains open.

If I change the `cat` command to `zsh` it obviously stays open side-by-side with the other shell. My question is - how do I ensure that the panel I want to run an arbitrary process from will remain open once the process completes - so I can re-use it myself if needed without seeing my layout broken?

I hope this makes sense.

**Comment by kovidgoyal** [comment, marked as answer]
launch sh -c "cat whatever; read"

**Reply by pm5k** [reply]
That's the bit I was unclear on. It seemed to me I could chain commands like so: `launch cat file && zsh` or  `launch cat file zsh`, thanks for the example.

**Reply by kovidgoyal** [reply]
I have added an example showing this to the launch documentation.
